static/js/markdown.js

```diff
@@ -74,11 +74,12 @@ exports.apply_markdown = function (message) {
             }
 
             if (person !== undefined) {
-                if (people.is_my_user_id(person.user_id)) {
+                if (people.is_my_user_id(person.user_id) && !silently) {
                     message.mentioned = true;
                     message.mentioned_me_directly = true;
                 }
-                return '<span class="user-mention" data-user-id="' + person.user_id + '">' +
+                var silent = silently ? '-silent' : '';
+                return '<span class="user-mention" data' + silent + '-user-id="' + person.user_id + '">' +
```

Comment:
This code is super not readable and not at all greppable for the `data-silent-user-id` key.  Just use an in-else statement for the full `<span>` construct if you're doing this sort of thing.

That said, I also think we should use the name `data-user-id` consistently, and just add a `silent` class if the mention is silent, rather than renaming data-user-id to data-silent-user-id.  Seems like a much cleaner interface.